build.sbt

```diff
@@ -159,7 +159,18 @@ scalacOptions in Compile in doc ++= Seq(
   "-diagrams-max-classes", "25",
   "-doc-version", version.value,
   "-doc-title", name.value,
-  "-doc-root-content", baseDirectory.value+"/root-doc.txt"
+  "-doc-root-content", baseDirectory.value+"/root-doc.txt",
+  "-sourcepath", (baseDirectory in ThisBuild).value.toString,
+  "-doc-source-url",
+  {
+    val branch =
+      if (version.value.endsWith("-SNAPSHOT")) {
+        "master"
+      } else {
+        s"v${version.value}"
+      }
+    s"https://github.com/freechipsproject/firrtl/tree/$branch€{FILE_PATH}.scala"
```

Comment:
Is `€` a euro sign...?

Reply:
I was surprised, too. It's the official escape character for that option :eu::

```
Usage: scaladoc <options> <source files>
where possible scaladoc options include:
  -doc-source-url <url>                       A URL pattern used to link to the source file; the following variables are available: €{TPL_NAME}, €{TPL_OWNER} and respectively €{FILE_PATH}. For example, for `scala.collection.Seq`, the variables will be expanded to `Seq`, `scala.collection` and respectively `scala/collection/Seq` (without the backquotes). To obtain a relative path for €{FILE_PATH} instead of an absolute one, use the -sourcepath setting.
```